checks/checkforcontent.go

```diff
@@ -78,10 +86,18 @@ func CheckFilesContent(checkName, shellPathFnPattern string, c *checker.CheckReq
 			continue
 		}
 
-		name := names[1]
-		// Filter out files based on path.
-		if match, _ := path.Match(shellPathFnPattern, name); !match {
-			continue
+		fullpath := names[1]
+		filename := path.Base(fullpath)
+
+		if !caseSensitive {
+			fullpath = strings.ToLower(fullpath)
+			filename = strings.ToLower(path.Base(fullpath))
+		}
+		// Filter out files based on path/names.
+		if match, _ := path.Match(shellPathFnPattern, fullpath); !match {
```

Comment:
Since, we only need the caseSensitive bool here for matching, let's do:

```
match := IsMatchingPath(pattern, path, filename, caseSensitive)
```

That way, the code is not littered with `if caseSensitive` and also the variable values for `shellPathFnPattern`, `fullpath` and `filename` remain the same inside this fn, irrespective of `caseSensitive`.

Reply:
you mean you want a new function called `IsMatchingPath`. Is that correct?

Reply:
Correct. Basically, that should (hopefully) be the only function whose logic depends on caseSensitive.